forui/lib/src/widgets/text_field/text_field.dart

```diff
@@ -983,6 +920,194 @@ class FTextField extends StatefulWidget {
+  ///     onPressed: () => controller.value = !controller.value,
+  ///     icon: Icon(controller.value ? Icons.visibility_off : Icons.visibility),
+  ///   ),
+  /// )
+  ///
+  /// // Disable toggle completely
+  /// FTextField.password(
+  ///   toggleBuilder: null,
+  /// )
+  /// ```
+  static Widget password({
+    FTextFieldStyle Function(FTextFieldStyle style)? style,
+    FFieldBuilder<FTextFieldStyle> builder = Defaults.builder,
+    Widget label = const Text('Password'),
+    String? hint,
+    Widget? description,
+    Widget? error,
+    TextMagnifierConfiguration? magnifierConfiguration,
+    Object groupId = EditableText,
+    TextEditingController? controller,
+    FocusNode? focusNode,
+    TextInputType? keyboardType,
+    TextInputAction textInputAction = TextInputAction.next,
+    TextCapitalization textCapitalization = TextCapitalization.none,
+    TextAlign textAlign = TextAlign.start,
+    TextAlignVertical? textAlignVertical,
+    TextDirection? textDirection,
+    bool autofocus = false,
+    WidgetStatesController? statesController,
+    String obscuringCharacter = '•',
+    bool autocorrect = false,
+    SmartDashesType? smartDashesType,
+    SmartQuotesType? smartQuotesType,
+    bool enableSuggestions = true,
+    int? minLines,
+    int maxLines = 1,
+    bool expands = false,
+    bool readOnly = false,
+    bool? showCursor,
+    int? maxLength,
+    MaxLengthEnforcement? maxLengthEnforcement,
+    ValueChanged<String>? onChange,
+    GestureTapCallback? onTap,
+    TapRegionCallback? onTapOutside,
+    bool onTapAlwaysCalled = false,
+    VoidCallback? onEditingComplete,
+    ValueChanged<String>? onSubmit,
+    AppPrivateCommandCallback? onAppPrivateCommand,
+    List<TextInputFormatter>? inputFormatters,
+    bool enabled = true,
+    bool? ignorePointers,
+    bool enableInteractiveSelection = true,
+    bool? selectAllOnFocus,
+    TextSelectionControls? selectionControls,
+    DragStartBehavior dragStartBehavior = DragStartBehavior.start,
+    MouseCursor? mouseCursor,
+    FTextFieldCounterBuilder? counterBuilder,
+    ScrollPhysics? scrollPhysics,
+    ScrollController? scrollController,
+    Iterable<String> autofillHints = const [AutofillHints.password],
+    String? restorationId,
+    bool stylusHandwritingEnabled = true,
+    bool enableIMEPersonalizedLearning = true,
+    ContentInsertionConfiguration? contentInsertionConfiguration,
+    EditableTextContextMenuBuilder contextMenuBuilder = Defaults.contextMenuBuilder,
+    bool canRequestFocus = true,
+    UndoHistoryController? undoController,
+    SpellCheckConfiguration? spellCheckConfiguration,
+    FFieldIconBuilder<FTextFieldStyle>? prefixBuilder,
+    FFieldIconBuilder<FTextFieldStyle>? suffixBuilder,
+    bool Function(TextEditingValue) clearable = Defaults.clearable,
+    String? initialText,
+    FPasswordToggleBuilder? toggleBuilder = _defaultToggleBuilder,
+    FValueNotifier<bool>? visibilityController,
```

Comment:
I think we can use a plain `ValueNotifier` here. Also I think we should rename this to `obscuredText`, kind of treating the notifier as a overload of the original `obscuredText`.